### PR TITLE
OSC output message formatting

### PR DESCRIPTION
**Pull request by jvlyndark**

### CODE REVIEW

File: autonomx/ComputeEngine.cpp [hunk 1]
```diff
@@ -182,16 +182,31 @@ void ComputeEngine::loop() {
...
             } else {
-                outputs.append(value);
+                oscOutputsList.append(value);
+                oscOutputRegion.append(value);
             }
+
+            // write and send individual output region osc messages
+            QString outputRegionNumber = QString::number(i+1);
+            oscOutputRegion.prepend("/" + (*it)->getGeneratorName() + "/output/" + outputRegionNumber);
+            emit sendOscData((*it)->getID(), oscOutputRegion);
```
Comment (aalex) [inline autonomx/ComputeEngine.cpp:202]: Can you add a comment saying what is the format of the OSC message(s) sent here?

File: autonomx/ComputeEngine.cpp [hunk 2]
```diff
@@ -182,16 +182,31 @@ void ComputeEngine::loop() {
...
+            oscOutputRegion.clear();
         }
-        emit sendOscData((*it)->getID(), outputs);
+
+        // write generator info to osc output list message
+        oscOutputsList.prepend("/" + (*it)->getGeneratorName() + "/output");
+
+        // send osc output list message
+        emit sendOscData((*it)->getID(), oscOutputsList);
```
Comment (aalex) [inline autonomx/ComputeEngine.cpp:211]: Can you add a comment saying what is the format of the OSC message(s) sent here?

**Review by aalex — CHANGES_REQUESTED** [review]
Can you add a comment saying what is the format of the OSC message(s) sent here?

**Review by aalex — APPROVED** [review]
LGTM

**Review by netherwaves — CHANGES_REQUESTED** [review]
will LGTM with this very very tiny change (sorry!)
- the packed OSC address should be `[...]/outputs` (with an "s" at the end) to distinguish it fully from the individual message address. this isn't critical but i think it clears a virtual point of confusion

**Comment by aalex** [comment]
Good idea. @jvlyndark Can you also make sure to edit the OSC.md file according to this new interface? (it can be done in another PR, if you prefer)

**Review by netherwaves — APPROVED** [review]
i've been convinced (also sorry)

**Pull request merged**